src/ASM/GlbL2projector.h

```diff
@@ -29,6 +30,75 @@ typedef std::vector<int>           IntVec;      //!< Vector of integers
 typedef std::vector<size_t>        uIntVec;     //!< Vector of unsigned integers
 typedef std::vector<FunctionBase*> FunctionVec; //!< Vector of functions
 
+/*!
+  \brief Abstract class for evaluating integrand or function.
+*/
+
+class L2Integrand {
+public:
+  //! \brief Constructor.
+  //! \param ASM ASM holding geometry to evaluate on
+  L2Integrand(const ASMbase& ASM) : m_asm(ASM) {}
```

Comment:
Here also, I don't think i like ASM as variable name, suggest `patch` or `pch`and then use `myPatch` (or `m_patch` if you prefer) as name on the `const ASMbase&` member